lib/state_machine/include/VehicleStateMachine.h

```diff
@@ -1,83 +1,70 @@
-    TRACTIVE_SYSTEM_ACTIVE = 2,
-    ENABLING_INVERTERS = 3,
-    WAITING_READY_TO_DRIVE_SOUND = 4,
-    READY_TO_DRIVE = 5
+enum class VehicleState_e {
+    TRACTIVE_SYSTEM_NOT_ACTIVE = 1, 
+    TRACTIVE_SYSTEM_ACTIVE = 2, 
+    WANTING_READY_TO_DRIVE = 3,
+    READY_TO_DRIVE = 4
 };
 
-/**
- * This singleton class represents the vehicle's state machine as we enable HV, enable the inverters,
- * wait for the start button to be pressed, and enter ready-to-drive mode. Aside from getters, this
- * class only has one public function, tick_state_machine(). In order to run its update logic, the
- * VehicleStateMachine uses instance data directly from other singleton classes (Buzzer, Pedals, etc)
- * and from the global structs (VCRInterfaceData_s, VCRData_s, etc.).
- */
 class VehicleStateMachine
 {
-public:
-    VehicleStateMachine(DrivetrainSystem & drivetrain_system) :
-        _current_state(CarState_e::STARTUP),
-        _drivetrain(drivetrain_system),
-        _buzzer(BuzzerController::getInstance()) {};
+    public: 
+        VehicleStateMachine(
+            etl::delegate<bool()> check_hv_over_threshold, 
+            etl::delegate<bool()> is_start_button_pressed, 
+            etl::delegate<bool()> is_brake_pressed, 
+            etl::delegate<bool()> check_drivetrain_error_ocurred, 
+            etl::delegate<bool()> check_drivetrain_ready,
+            etl::delegate<void()> start_buzzer,
+            etl::delegate<bool()> is_buzzer_done, 
+            etl::delegate<void()> end_buzzer, 
+            etl::delegate<void()> handle_drivetrain_init, 
+            etl::delegate<void()> command_drivetrain
+        ) :  
+        _check_hv_over_threshold(check_hv_over_threshold),
+        _is_start_button_pressed(is_start_button_pressed), 
+        _is_brake_pressed(is_brake_pressed),
+        _check_drivetrain_error_ocurred(check_drivetrain_error_ocurred),
+        _check_drivetrain_ready(check_drivetrain_ready),
+        _start_buzzer(start_buzzer),
+        _is_buzzer_complete(is_buzzer_done),
+        _end_buzzer(end_buzzer),
+        _handle_drivetrain_init(handle_drivetrain_init),
+        _command_drivetrain(command_drivetrain)
+        {   
+            _current_state = VehicleState_e::TRACTIVE_SYSTEM_NOT_ACTIVE;
+        }
 
-    VehicleStateMachine& operator & (VehicleStateMachine& ) = delete;
-    /**
-     * This tick() function handles all the update logic for traversing states, and calls the functions
-     * of other classes as necessary.
-     * @pre Other systems are updated properly
-     * @pre All relevant data exists in the data structs (VCRInterfaceData, VCRSystemData, etc.)
-     * @param current_millis The system time, in millis. Passed in by the scheduler.
-     * @param system_data A reference to the global system data struct.
-     */
-    void tick_state_machine(unsigned long current_millis, const VCRData_s &system_data);
+        void tick_state_machine(unsigned long curr_time_millis);
```

Comment:
Is this better implemented as separate tick() and get_state() functions, or should tick() return the next VehicleState_e? The feature/drivebrain_interface_and_system PR expects the VSM to return the next car state. I don't see a good reason to split the two, other than for easier unit testing.

Reply:
Yeah, I agree that it's better as one. It currently is like that, I just changed the signature because it doesn't depend on the VCR Data struct anymore.